diff --git a/kernel/Kconfig b/kernel/Kconfig
index 816905a1..d28cc056 100644
--- a/kernel/Kconfig
+++ b/kernel/Kconfig
@@ -13,7 +13,8 @@ config KSU_WITH_KPROBES
 	bool "Use kprobes for kernelsu"
 	depends on KSU
 	depends on KPROBES
-	default y
+	depends on BROKEN
+	default n
 	help
 	  Disable if you use manual hooks.
 
@@ -51,7 +52,7 @@ config KSU_SUSFS_HAS_MAGIC_MOUNT
 config KSU_SUSFS_SUS_PATH
     bool "Enable to hide suspicious path (NOT recommended)"
     depends on KSU_SUSFS
-    default y
+    default n
     help
       - Allow hiding the user-defined path and all its sub-paths from various system calls.
       - tmpfs filesystem is not allowed to be added. 
@@ -164,8 +165,8 @@ config KSU_SUSFS_OPEN_REDIRECT
 
 config KSU_SUSFS_SUS_SU
     bool "Enable SUS-SU in runtime temporarily"
-    depends on KSU_SUSFS && KPROBES && HAVE_KPROBES && KPROBE_EVENTS
-    default y
+    depends on KSU_SUSFS && KPROBES && HAVE_KPROBES && KPROBE_EVENTS && BROKEN
+    default n
     help
       - Allow user to enable or disable core ksu kprobes hooks temporarily in runtime. There are 2 working modes for sus_su.
       - Mode 0 (default): Disable sus_su, and enable ksu kprobe hooks for su instead.
